### DISCUSSION TITLE
Configuring connection strings for the test

### DISCUSSION THREAD

**Discussion by ndc** [xunit/xunit#2995] [Question] [answered]
I want to supply a connection string to my tests, but I want to specify different connection string if the test is run in the build server. What is the best practice to do that?
I tried using Microsoft.Extensions.Configuration.EnvironmentVariables' ConfigurationBuilder. I added the environment variable in Visual Studio properties - debug - launch profile (the launchSettings.json). But looks like Visual Studio's Test Explorer ignores this launchSettings.json and I didn't get the connection string when running the tests through it.
So I ended up embedding the connection string into xunit.runner.json since it is already there so I don't have to add another config file. This contains the connection string for local development.
To run the test from my Azure DevOps pipeline I plan to add environment variable to override the one in xunit.runner.json. I haven't gone this far though.
Will I encounter problem with this plan?

**Comment by bradwilson** [comment, marked as answer]
Sorry for the delay in answering.

> Will I encounter problem with this plan?

No, this won't be a problem, so long as you pick a key name that doesn't collide with the names we're already using.
